src/parser.rs

```diff
@@ -37,21 +39,37 @@ pub trait ArgumentChecker<C>: Any + Send + Sync + 'static {
     fn box_clone(&self) -> Box<dyn ArgumentChecker<C>>;
 }
 
-pub trait ArgumentParser<C: Context>: Send + Sync + 'static {
-    type Output: Send + Sync;
+pub trait ArgumentParser<C: Context>: Send + 'static {
+    type Output: Send;
 
     fn parse<'a, 'b>(
         &self,
         ctx: &mut C,
         input: &'a mut &'b str,
-    ) -> Pin<Box<dyn Future<Output = anyhow::Result<Self::Output>> + Send + 'a>>;
+    ) -> FutureBox<'a, Result<Self::Output, C::Err>>;
     fn default() -> Self
     where
         Self: Sized;
 }
 
-pub trait ArgumentKind<C: Context>: Sized + Send + Sync {
+pub trait ArgumentSuggester<C>
+where
+    C: Context,
+{
+    fn suggestions<'a, 'b, 'c>(&'a self, _ctx: &'b C, _input: &'c str) -> FutureBox<'c, Vec<String>>;
```

Comment:
Could the return value be `Cow<'static, str>` instead, to reduce allocations in trivial cases?